src/ergw_aaa_gx.erl

```diff
@@ -299,6 +299,10 @@ handle_cca({error, no_connection}, Session, Events,
 	   #{answer_if_timeout := Answer, answers := Answers} = Opts, State) ->
     Avps = apply_answer_config(Answer, Answers),
     handle_cca(['CCA' | Avps], Session, Events, Opts, State);
+handle_cca({error, Code} = Result, Session, Events, _Opts, State)
+  when (Code >= 3000 andalso Code < 4000) orelse
+       (Code >= 5000 andalso Code < 6000) ->
```

Comment:
this is somewhat redundant. Only results codes in the 3000-4000 and 5000-6000 range should be sent with E bit set and would therefore result in an `{error, Cause}` response.

Checking for integer would also make sure that we catch all cases when a result code sneaked into the response:
```suggestion
  when is_integer(Code) ->
```

Reply:
Ok